cartoframes/data/dataset/registry/dataframe_dataset.py

```diff
@@ -53,14 +52,14 @@ def download(self, limit, decode_geom, retry_times):
     def upload(self, if_exists, with_lnglat):
         self._is_ready_for_upload_validation()
 
-        normalized_column_names = _normalize_column_names(self._df)
+        columns, geom_column, enc_type = _process_columns(self._df, with_lnglat)
```

Comment:
This is fine since it's just a private method, kind of an utility. But I find difficult to understand methods that return multiple values... you have to remember the order or maybe you don't need all the variables, etc.

Again it's fine as long as it's private 👍

Reply:
Yes, it is a bit painful, but it is used once and very ad hoc to the current problem. It is difficult to split the problem into different methods because they are very coupled.